code/modules/power/supermatter/supermatter_hit_procs.dm

```diff
@@ -44,6 +32,22 @@
 			visible_message(span_notice("[src] compresses under stress, resisting further impacts!"))
 	return BULLET_ACT_HIT
 
+/obj/machinery/power/supermatter_crystal/proc/log_activation(source, cause)
+	var/fired_from_str = cause ? " with [cause]" : ""
+	investigate_log(
+		source \
+			? "has been powered for the first time by [key_name(source)][fired_from_str]." \
+			: "has been powered for the first time.",
+		INVESTIGATE_ENGINE
+	)
+	message_admins(
+		source \
+			? "[src] [ADMIN_JMP(src)] has been powered for the first time by [cause ? ADMIN_FULLMONTY(source) + (fired_from_str) : (source ? source : "environmental factors")]." \
```

Comment:
why the second source ? : here?

Reply:
if a source is not passed or non-atomic it falls back. so theres always something if any combo happens

Reply:
yeah but we can't get to this branch of logic unless there's a source, right?

Reply:
oh right maybe I intended it to be a cause

Reply:
ima bap this real fast, in any case the messages that are spat out are human readable and make sense